## BIG 5 ANSWERS EXPLORATION & CLARIFICATIONS

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
questions = pd.read_csv('../csv/questions.csv')
responses = pd.read_csv('../csv/responses.csv')

In [4]:
questions[questions['category']=='Personality'][['subcategory','prompt']]

subcategory                           prompt
1             Reflection      Enjoy the beauty of nature.
2    Emotional Stability    Feel comfortable with myself.
4                Anxiety          Feel threatened easily.
5        Resourcefulness  Like to solve complex problems.
6                Empathy       Love to reflect on things.
..                   ...                              ...
146           Efficiency              Postpone decisions.
147       Purposefulness            Am easily distracted.
148         Extraversion           Hold back my opinions.
149              Empathy          Don't have a soft side.
150    Humor/Playfulness               Have a lot of fun.

[100 rows x 2 columns]

In [5]:
len(questions)

151

So we have 151 different BIG 5 questions. Let's check how many questions are in each Personality feature

In [22]:
questions[questions['category']=='Personality']['subcategory'].unique()

array(['Reflection', 'Emotional Stability', 'Anxiety', 'Resourcefulness',
       'Empathy', 'Positive Expressivity', 'Assertiveness',
       'Impulse-Control', 'Orderliness', 'Efficiency', 'Agreeableness',
       'Calmness', 'Conscientiousness', 'Imagination',
       'Vitality/Enthusiasm/Zest', 'Anger', 'Warmth', 'Intellect',
       'Industriousness/Perseverance/Persistence', 'Sympathy',
       'Cooperation', 'Emotionality', 'Understanding', 'Methodicalness',
       'Reserve', 'Leadership', 'Complexity', 'Stability', 'Sensitivity',
       'Responsive Joy', 'Aesthetic Appreciation/Artistic Interests',
       'Attractiveness', 'Comprehension', 'Extraversion', 'Poise',
       'Tranquility', 'Perfectionism', 'Purposefulness', 'Nurturance',
       'Quickness', 'Dutifulness', 'Achievement-Striving',
       'Gregariousness', 'Happiness', 'Provocativeness', 'Friendliness',
       'Toughness', 'Humor/Playfulness'], dtype=object)

In [24]:
questions[questions['category']=='Personality']['subcategory'].nunique()

48

In [6]:
personality_subcategories = questions[questions['category'] == 'Personality']['subcategory'].unique()

formatted_list = "\n".join([f"- {subcategory}" for subcategory in personality_subcategories])

print("Personality Subcategories:")
print(formatted_list)

Personality Subcategories:
- Reflection
- Emotional Stability
- Anxiety
- Resourcefulness
- Empathy
- Positive Expressivity
- Assertiveness
- Impulse-Control
- Orderliness
- Efficiency
- Agreeableness
- Calmness
- Conscientiousness
- Imagination
- Vitality/Enthusiasm/Zest
- Anger
- Warmth
- Intellect
- Industriousness/Perseverance/Persistence
- Sympathy
- Cooperation
- Emotionality
- Understanding
- Methodicalness
- Reserve
- Leadership
- Complexity
- Stability
- Sensitivity
- Responsive Joy
- Aesthetic Appreciation/Artistic Interests
- Attractiveness
- Comprehension
- Extraversion
- Poise
- Tranquility
- Perfectionism
- Purposefulness
- Nurturance
- Quickness
- Dutifulness
- Achievement-Striving
- Gregariousness
- Happiness
- Provocativeness
- Friendliness
- Toughness
- Humor/Playfulness


In [45]:
# Filter the subcategories from the 'questions' DataFrame
personality_subcategories = questions[questions['category'] == 'Personality']['subcategory'].unique()

# Create a list of Big Five categories corresponding to each subcategory
big_five_categories = [
    'Openness', 'Neuroticism', 'Neuroticism', 'Conscientiousness', 'Agreeableness', 'Extraversion', 'Extraversion', 'Conscientiousness',
    'Neuroticism', 'Conscientiousness', 'Agreeableness', 'Neuroticism', 'Conscientiousness', 'Openness', 'Extraversion', 'Neuroticism', 'Agreeableness',
    'Openness', 'Conscientiousness', 'Agreeableness', 'Agreeableness', 'Neuroticism', 'Agreeableness', 'Conscientiousness', 'Extraversion',
    'Extraversion', 'Openness', 'Neuroticism', 'Neuroticism', 'Extraversion', 'Openness', 'Openness', 'Openness', 'Extraversion', 'Extraversion', 'Neuroticism',
    'Conscientiousness', 'Conscientiousness', 'Conscientiousness', 'Openness', 'Conscientiousness', 'Conscientiousness', 'Extraversion', 'Extraversion',
    'Extraversion', 'Extraversion', 'Neuroticism', 'Extraversion'
]

# Create a DataFrame
big5 = pd.DataFrame({'Subcategory': personality_subcategories, 'Category': big_five_categories})




In [46]:
big5

Subcategory           Category
0                                  Reflection           Openness
1                         Emotional Stability        Neuroticism
2                                     Anxiety        Neuroticism
3                             Resourcefulness  Conscientiousness
4                                     Empathy      Agreeableness
5                       Positive Expressivity       Extraversion
6                               Assertiveness       Extraversion
7                             Impulse-Control  Conscientiousness
8                                 Orderliness        Neuroticism
9                                  Efficiency  Conscientiousness
10                              Agreeableness      Agreeableness
11                                   Calmness        Neuroticism
12                          Conscientiousness  Conscientiousness
13                                Imagination           Openness
14                   Vitality/Enthusiasm/Zest       Extraversion
15                                      Anger        Neuroticism
16                                     Warmth      Agreeableness
17                                  Intellect           Openness
18   Industriousness/Perseverance/Persistence  Conscientiousness
19                                   Sympathy      Agreeableness
20                                Cooperation      Agreeableness
21                               Emotionality        Neuroticism
22                              Understanding      Agreeableness
23                             Methodicalness  Conscientiousness
24                                    Reserve       Extraversion
25                                 Leadership       Extraversion
26                                 Complexity           Openness
27                                  Stability        Neuroticism
28                                Sensitivity        Neuroticism
29                             Responsive Joy       Extraversion
30  Aesthetic Appreciation/Artistic Interests           Openness
31                             Attractiveness           Openness
32                              Comprehension           Openness
33                               Extraversion       Extraversion
34                                      Poise       Extraversion
35                                Tranquility        Neuroticism
36                              Perfectionism  Conscientiousness
37                             Purposefulness  Conscientiousness
38                                 Nurturance  Conscientiousness
39                                  Quickness           Openness
40                                Dutifulness  Conscientiousness
41                       Achievement-Striving  Conscientiousness
42                             Gregariousness       Extraversion
43                                  Happiness       Extraversion
44                            Provocativeness       Extraversion
45                               Friendliness       Extraversion
46                                  Toughness        Neuroticism
47                          Humor/Playfulness       Extraversion

In [47]:
big5['Category'].unique()

array(['Openness', 'Neuroticism', 'Conscientiousness', 'Agreeableness',
       'Extraversion'], dtype=object)

In [48]:
big5[big5['Category']== 'Openness']

Subcategory  Category
0                                  Reflection  Openness
13                                Imagination  Openness
17                                  Intellect  Openness
26                                 Complexity  Openness
30  Aesthetic Appreciation/Artistic Interests  Openness
31                             Attractiveness  Openness
32                              Comprehension  Openness
39                                  Quickness  Openness

Reflection belongs here. Where do we place Attractiveness?? Intellect is OK

In [49]:
big5[big5['Category']== 'Conscientiousness']

Subcategory           Category
3                            Resourcefulness  Conscientiousness
7                            Impulse-Control  Conscientiousness
9                                 Efficiency  Conscientiousness
12                         Conscientiousness  Conscientiousness
18  Industriousness/Perseverance/Persistence  Conscientiousness
23                            Methodicalness  Conscientiousness
36                             Perfectionism  Conscientiousness
37                            Purposefulness  Conscientiousness
38                                Nurturance  Conscientiousness
40                               Dutifulness  Conscientiousness
41                      Achievement-Striving  Conscientiousness

Impulse control belongs to Neuroticism

In [50]:
big5[big5['Category']== 'Extraversion']

Subcategory      Category
5      Positive Expressivity  Extraversion
6              Assertiveness  Extraversion
14  Vitality/Enthusiasm/Zest  Extraversion
24                   Reserve  Extraversion
25                Leadership  Extraversion
29            Responsive Joy  Extraversion
33              Extraversion  Extraversion
34                     Poise  Extraversion
42            Gregariousness  Extraversion
43                 Happiness  Extraversion
44           Provocativeness  Extraversion
45              Friendliness  Extraversion
47         Humor/Playfulness  Extraversion

Provocativeness does belong to this category (assertiveness). Positive Expressivity and Friendliness belong here


#### Maybe some questions belong to different categories? NO. too complex
#### How do big5 test scores work https://www.jordanbpeterson.com/docs/230/2014/15DeYoung.pdf
Some questions are positive and others are reversed (all of them score 1-5 / 1-3)

In [51]:
big5[big5['Category']== 'Agreeableness']

Subcategory       Category
4         Empathy  Agreeableness
10  Agreeableness  Agreeableness
16         Warmth  Agreeableness
19       Sympathy  Agreeableness
20    Cooperation  Agreeableness
22  Understanding  Agreeableness

Quickness belongs to the 'Openness' category (Intellect, Jordan Peterson), and Gregariousness belongs to the 'Extraversion' category (J. Peterson).
Warmth belongs to 'Agreeableness' but also to 'Extraversion' ??? what should we consider then?

In [52]:
big5[big5['Category']== 'Neuroticism']

Subcategory     Category
1   Emotional Stability  Neuroticism
2               Anxiety  Neuroticism
8           Orderliness  Neuroticism
11             Calmness  Neuroticism
15                Anger  Neuroticism
21         Emotionality  Neuroticism
27            Stability  Neuroticism
28          Sensitivity  Neuroticism
35          Tranquility  Neuroticism
46            Toughness  Neuroticism

It's OK

In [53]:
big5[['Subcategory','Category']]

Subcategory           Category
0                                  Reflection           Openness
1                         Emotional Stability        Neuroticism
2                                     Anxiety        Neuroticism
3                             Resourcefulness  Conscientiousness
4                                     Empathy      Agreeableness
5                       Positive Expressivity       Extraversion
6                               Assertiveness       Extraversion
7                             Impulse-Control  Conscientiousness
8                                 Orderliness        Neuroticism
9                                  Efficiency  Conscientiousness
10                              Agreeableness      Agreeableness
11                                   Calmness        Neuroticism
12                          Conscientiousness  Conscientiousness
13                                Imagination           Openness
14                   Vitality/Enthusiasm/Zest       Extraversion
15                                      Anger        Neuroticism
16                                     Warmth      Agreeableness
17                                  Intellect           Openness
18   Industriousness/Perseverance/Persistence  Conscientiousness
19                                   Sympathy      Agreeableness
20                                Cooperation      Agreeableness
21                               Emotionality        Neuroticism
22                              Understanding      Agreeableness
23                             Methodicalness  Conscientiousness
24                                    Reserve       Extraversion
25                                 Leadership       Extraversion
26                                 Complexity           Openness
27                                  Stability        Neuroticism
28                                Sensitivity        Neuroticism
29                             Responsive Joy       Extraversion
30  Aesthetic Appreciation/Artistic Interests           Openness
31                             Attractiveness           Openness
32                              Comprehension           Openness
33                               Extraversion       Extraversion
34                                      Poise       Extraversion
35                                Tranquility        Neuroticism
36                              Perfectionism  Conscientiousness
37                             Purposefulness  Conscientiousness
38                                 Nurturance  Conscientiousness
39                                  Quickness           Openness
40                                Dutifulness  Conscientiousness
41                       Achievement-Striving  Conscientiousness
42                             Gregariousness       Extraversion
43                                  Happiness       Extraversion
44                            Provocativeness       Extraversion
45                               Friendliness       Extraversion
46                                  Toughness        Neuroticism
47                          Humor/Playfulness       Extraversion

In [54]:
# Group by 'Category' and count the number of subcategories in each category
summary = big5.groupby('Category')['Subcategory'].count().reset_index()

# Display the summary
print(summary)

# List the subcategories for each Big Five category
for category, subcategories in big5.groupby('Category')['Subcategory']:
    print(f"Subcategories in {category} category:")
    print(subcategories.tolist())
    print()

            Category  Subcategory
0      Agreeableness            6
1  Conscientiousness           11
2       Extraversion           13
3        Neuroticism           10
4           Openness            8
Subcategories in Agreeableness category:
['Empathy', 'Agreeableness', 'Warmth', 'Sympathy', 'Cooperation', 'Understanding']

Subcategories in Conscientiousness category:
['Resourcefulness', 'Impulse-Control', 'Efficiency', 'Conscientiousness', 'Industriousness/Perseverance/Persistence', 'Methodicalness', 'Perfectionism', 'Purposefulness', 'Nurturance', 'Dutifulness', 'Achievement-Striving']

Subcategories in Extraversion category:
['Positive Expressivity', 'Assertiveness', 'Vitality/Enthusiasm/Zest', 'Reserve', 'Leadership', 'Responsive Joy', 'Extraversion', 'Poise', 'Gregariousness', 'Happiness', 'Provocativeness', 'Friendliness', 'Humor/Playfulness']

Subcategories in Neuroticism category:
['Emotional Stability', 'Anxiety', 'Orderliness', 'Calmness', 'Anger', 'Emotionality', 'Stabili

## SOMETHING IS WRONG
#### Big 5 scores don't add up

The scores of the answers go from 1 to 3 (or are we considering 1-5 when adding?) in the big 5 answers. So the minimum and maximum should be:

-       Agreeableness           8-24
-       Conscientiousness       12-36
-       Extraversion            11-33
-       Neuroticism             9-27
-       Openness                8-24

But the ranges of values we have in the participants' responses are: (CALCULATED IN Exploratory.ipynb)

-       Agreeableness 	    24-42
-       Conscientiousness  	    19-45
-       Extraversion 	            17-42
-       Neuroticism 	            12-36
-       Openness 	            16-30

If we are considering scores from 1 to 5, then minimum and maximum should be:

-       Agreeableness           8-40
-       Conscientiousness       12-60
-       Extraversion            11-55
-       Neuroticism             9-45
-       Openness                8-40

But this doesn't make sense, since we have at least one response with a score of 42 in agreeableness. I'm missing something or something is not working...

### BOBO!! estas asumiendo que hay una sola pregunta por subcategoría, pero realmente hay mas de 3... recuerda que al principio viste que había 150 preguntas...